server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -197,8 +198,10 @@ private void assertMappingVersion(
                     assert currentSource.equals(newSource) :
```

Comment:
Do we need this assertion as well? isn't it covered now by asserting the maps?

Reply:
It's a bit different - there's currentSorce, newSource and mapping.source.

Given that it currently doesn't cause issues I left it as it's. This will likely all disappear anyway as part of the first class tables work